### ISSUE TITLE
"Insufficient Funds" when Primary Currency is set to Fiat

### ISSUE THREAD

**Issue by deshvin** [MetaMask/metamask-mobile#5737]
**Describe the bug**
When the setting **Settings** > **General** > **Primary Currency** is set to **Fiat**, sending a USD-value amount of a token that is above the number held of that token results in an "Insufficient Funds" error message.

**Screenshots**
https://user-images.githubusercontent.com/2859402/217675235-520e3111-e1ff-413a-939b-3c45fc0ee6bb.mov

**To Reproduce**
_Steps to reproduce the behavior_
1. change **Settings** > **General** > **Primary Currency** to **Fiat**
2. Try to Send a token whose fiat currency value is higher than the number of the token held
3. Select MAX or enter a USD value higher than the amount of the token held
4. Select Next
5. "Insufficient funds"

Tested on BSC and Avalanche. Issues does **not** seem to occur on Ethereum Mainnet.

**Expected behavior**
I should be able to send tokens when my primary currency setting is set to Fiat

**Smartphone (please complete the following information):**
 - Device: MetaMask in AppStore on MacOS, though have confirmed this also happens in Android
 - OS: 
 - App Version: 5.14.0

-------------------------------------------------------------
_to be added after bug submission by internal support / PM_
**Severity**

- How critical is the impact of this bug on a user?
- Add stats if available on % of customers impacted
- Is this visible to all users?
- Is this tech debt?

**Comment by CharlesBlaqk** [comment]
@deshvin We'd be happy to look into this for you. kindly file a support ticket below 
https://forms.gle/Wmpm6NLzYm9uXYuU9

**Comment by bschorchit** [comment]
It seems to be an issue with how we're handling the conversion. When I first tried, reproducing it I saw no insufficient funds warning, but clicking on max would display the max amount in token next to the $ symbol instead of the max amount in fiat (screenshot).

When I tried again a few minutes later, I had the same experience as described in the issue and saw the no insufficient funds warning.

![IMG_3652](https://user-images.githubusercontent.com/34968356/217798855-3682b016-c7a8-4a52-8025-10ea333b238d.PNG)
